### PR TITLE
tikv: add new dynamic configure for flow-control

### PR DESCRIPTION
**Pull request by glorv**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/18540
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by Connor1996 — APPROVED** [review]
LGTM

File: dynamic-config.md [hunk 1]
```diff
@@ -225,6 +225,11 @@ The following TiKV configuration items can be modified dynamically:
 | `server.raft-msg-max-batch-size` | Sets the maximum number of Raft messages that are contained in a single gRPC message |
 | `server.simplify-metrics`        | Controls whether to simplify the sampling monitoring metrics                   |
 | `storage.block-cache.capacity` | The size of shared block cache (supported since v4.0.3) |
+| storage.flow-control.enable | Determines whether to enable the flow control mechanism. |
+| storage.flow-control.memtables-threshold | The threshold of kvDB memtable numbers that triggers flow control. |
+| storage.flow-control.l0-files-threshold | The threshold of kvDB L0 file numbers that triggers flow control. |
+| storage.flow-control.soft-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control. |
+| storage.flow-control.hard-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control mechanism to reject all write requests and reports the `ServerIsBusy` error. |
```
Comment (Oreoxmt) [inline dynamic-config.md:232]:
```suggestion
| storage.flow-control.enable | Determines whether to enable the flow control mechanism |
| storage.flow-control.memtables-threshold | The threshold of kvDB memtable numbers that triggers flow control |
| storage.flow-control.l0-files-threshold | The threshold of kvDB L0 file numbers that triggers flow control |
| storage.flow-control.soft-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control |
| storage.flow-control.hard-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control mechanism to reject all write requests and report the `ServerIsBusy` error |
```

Keep the format consistent with other rows in this table.

**Review by Oreoxmt — APPROVED** [review]

File: dynamic-config.md [hunk 2]
```diff
@@ -225,6 +225,11 @@ The following TiKV configuration items can be modified dynamically:
 | `server.raft-msg-max-batch-size` | Sets the maximum number of Raft messages that are contained in a single gRPC message |
 | `server.simplify-metrics`        | Controls whether to simplify the sampling monitoring metrics                   |
 | `storage.block-cache.capacity` | The size of shared block cache (supported since v4.0.3) |
+| storage.flow-control.enable | Determines whether to enable the flow control mechanism |
+| storage.flow-control.memtables-threshold | The threshold of kvDB memtable numbers that triggers flow control |
+| storage.flow-control.l0-files-threshold | The threshold of kvDB L0 file numbers that triggers flow control |
+| storage.flow-control.soft-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control |
+| storage.flow-control.hard-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control mechanism to reject all write requests and report the `ServerIsBusy` error |
```
Comment (qiancai) [inline dynamic-config.md:232]:
```suggestion
| storage.flow-control.hard-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control mechanism to reject all write requests |
```

Comment (qiancai) [reply dynamic-config.md:232]: keep it consistent with the Chinese PR: https://github.com/pingcap/docs-cn/pull/18540/files

File: dynamic-config.md [hunk 3]
```diff
@@ -225,6 +225,11 @@ The following TiKV configuration items can be modified dynamically:
 | `server.raft-msg-max-batch-size` | Sets the maximum number of Raft messages that are contained in a single gRPC message |
 | `server.simplify-metrics`        | Controls whether to simplify the sampling monitoring metrics                   |
 | `storage.block-cache.capacity` | The size of shared block cache (supported since v4.0.3) |
+| storage.flow-control.enable | Determines whether to enable the flow control mechanism |
+| storage.flow-control.memtables-threshold | The threshold of kvDB memtable numbers that triggers flow control |
+| storage.flow-control.l0-files-threshold | The threshold of kvDB L0 file numbers that triggers flow control |
+| storage.flow-control.soft-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control |
```
Comment (qiancai) [inline dynamic-config.md:231]:
```suggestion
| storage.flow-control.soft-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control mechanism to reject some write requests |
```

File: dynamic-config.md [hunk 4]
```diff
@@ -225,6 +225,11 @@ The following TiKV configuration items can be modified dynamically:
 | `server.raft-msg-max-batch-size` | Sets the maximum number of Raft messages that are contained in a single gRPC message |
 | `server.simplify-metrics`        | Controls whether to simplify the sampling monitoring metrics                   |
 | `storage.block-cache.capacity` | The size of shared block cache (supported since v4.0.3) |
+| storage.flow-control.enable | Determines whether to enable the flow control mechanism |
+| storage.flow-control.memtables-threshold | The threshold of kvDB memtable numbers that triggers flow control |
+| storage.flow-control.l0-files-threshold | The threshold of kvDB L0 file numbers that triggers flow control |
```
Comment (qiancai) [inline dynamic-config.md:230]:
```suggestion
| storage.flow-control.memtables-threshold | The maximum number of kvDB memtables that trigger flow control |
| storage.flow-control.l0-files-threshold | The maximum number of kvDB L0 files that trigger flow control |
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Review by qiancai — APPROVED** [review]

**Pull request merged**